File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.cs
public class Solution {
    public int FindMin(int[] nums) {
        var l = 0;
        var r = nums.Length - 1;
        
        while (l < r)
        {
            var lV = nums[l];
            var rV = nums[r];

            var m = (r - l) / 2 + l;
            var mV = nums[m];

            if (mV > rV) l = m + 1;
            else r = m;
        }

        return nums[l];
    }
}

File: Data Structures & Algorithms/find-duplicate-integer/submission-1.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        # O(n) space
        counter = [0] * (len(nums) + 1)

        for n in nums:
            counter[n] += 1
            if counter[n] > 1:
                return n


